src/esn.calendar.libs/app/components/event-recurrence-edition/event-recurrence-edition.spec.js

```diff
@@ -45,115 +51,125 @@ describe('The event-recurrence-edition component', function() {
 
     this.initDirective = function(scope) {
       var html = '<event-recurrence-edition event="event" can-modify-event-recurrence="canModifyEventRecurrence"/>';
-      var element = this.$compile(html)(scope);
-
-      scope.$digest();
-      this.eleScope = element.isolateScope();
-
-      return element;
+      var element = self.$compile(html)(scope);
+
+      return new Promise(function(resolve) {
+        setTimeout(function() {
+          scope.$digest();
+          self.eleScope = element.isolateScope();
+          return resolve(element);
+        }, 0);
+      });
     };
   }]));
 
   describe('activate function', function() {
-    it('should set days correctly from input event', function() {
+    it('should set days correctly from input event', function(done) {
       this.$scope.event.rrule = {
         freq: undefined,
         interval: null,
         byday: ['SA', 'SU']
       };
-      this.initDirective(this.$scope);
-
-      expect(this.eleScope.vm.days).to.shallowDeepEqual([
-        {value: 'MO', selected: false},
-        {value: 'TU', selected: false},
-        {value: 'WE', selected: false},
-        {value: 'TH', selected: false},
-        {value: 'FR', selected: false},
-        {value: 'SA', selected: true},
-        {value: 'SU', selected: true}
-      ]);
+      this.initDirective(this.$scope).then(function(){
+        expect(self.eleScope.vm.days).to.shallowDeepEqual([
+          {value: 'MO', selected: false},
+          {value: 'TU', selected: false},
+          {value: 'WE', selected: false},
+          {value: 'TH', selected: false},
+          {value: 'FR', selected: false},
+          {value: 'SA', selected: true},
+          {value: 'SU', selected: true}
+        ]);
+      }).then(done);
```

Comment:
Please  `catch` all the errors in this PR to have clear logs

```suggestion
      }).then(done)
      .catch(err => done(err || new Error('should resolve'));
```